src/utils/sockctl.c

```diff
@@ -23,13 +23,83 @@
 #include "net.h"
 
 #define SOCK_EXTENSION ".sock"
-
+#define TMP_UNIX_SOCK_FOLDER_TEMPLATE "/tmp/edgesec.XXXXXX"
+#define TMP_UNIX_SOCK_NAME "client-socket" SOCK_EXTENSION
 #define DOMAIN_REPLY_TIMEOUT 10
 
 void init_domain_addr(struct sockaddr_un *unaddr, const char *addr) {
   *unaddr = (struct sockaddr_un){.sun_family = AF_UNIX};
   os_strlcpy(unaddr->sun_path, addr, sizeof(unaddr->sun_path));
 }
+struct tmp_domain_socket {
```

Comment:
Could you please explain this structure?

Reply:
How are these doxygen docs? (added in https://github.com/nqminds/edgesec/pull/278/commits/b90bc2dd16e7b9ba430420a2dd79b1c7ae4355f7)

Essentially, it's a hashmap struct that stores all the temporary unix domain sockets (and the folder they are in) created by `create_tmp_domain_socket_path()` so that the sockets and the folders can be properly deleted by `cleanup_tmp_domain_socket_path()`

https://github.com/nqminds/edgesec/blob/b90bc2dd16e7b9ba430420a2dd79b1c7ae4355f7/src/utils/sockctl.c#L38-L54

Reply:
We are using here a global variable. In general I prefer not to use any in the edgesec code. I think we should refactor and add things like context or similar. There's also the problem of thread safety. We haven't done any locking for the hash string. Check the thread safety section in https://troydhanson.github.io/uthash/userguide.html

Reply:
> We are using here a global variable. In general I prefer not to use any in the edgesec code.

Ahhh, that will make things difficult. I agree, global variables are a nightmare. I only implemented it that way to change as little as I could as possible.

It's a pity that in C you often have to use global variables, especially for stuff like signal handling.

Hmmmmm, in that case, I need to think of the best way of how to handle cleanup. How about:
  - Return a `struct unix_socket {int fd; (void *)(void) destructor;}` from `create_domain_client()`, so we can return a custom `destructor()` function (C++ style)
  - Just ignore cleaning up the domain sockets.
    - They're only used in FreeBSD
    - If we stick them all into a `/tmp/edgesec/tmp-sockets.XXXXXX/*`, we can just not bother cleaning them up, and as long it takes at years for 1 million tmp files to be created, FreeBSD will eventually shutdown and delete the `/tmp` folder.

Reply:
I think the best answer is to ignore cleaning up the tmp files.

Reply:
Could you please add a `log_warn` or similar saying which files are being deleted?

Reply:
Added in https://github.com/nqminds/edgesec/pull/278/commits/ae16f1c00642f322e1b870a48f8100ea833c7422

I've made it is only a `log_debug()`. This is because if we do try to delete the wrong folder, chances are that it's not empty and so `rmdir()` will have a `ENOTEMPTY` (dir not empty) error, and we'll get a `log_errno()` message.